core/src/test/java/org/infinispan/persistence/ParallelIterationTest.java

```diff
@@ -111,18 +116,20 @@ public void testCancelingTaskMultipleProcessors() {
       final ConcurrentMap<Object, Object> entries = new ConcurrentHashMap<>();
       final AtomicBoolean stopped = new AtomicBoolean(false);
 
-      loader.process(null, (marshalledEntry, taskContext) -> {
-         synchronized (entries) {
-            boolean shouldStop = entries.size() == 100 && !stopped.get();
-            log.trace("shouldStop = " + shouldStop + ",entries size = " + entries.size());
-            if (shouldStop) {
-               stopped.set(true);
-               taskContext.stop();
-               return;
-            }
-            entries.put(unwrapKey(marshalledEntry.getKey()), unwrapValue(marshalledEntry.getValue()));
-         }
-      }, executor, true, true);
+      Flowable.fromPublisher(loader.publishEntries(null, true, true))
+            .observeOn(Schedulers.from(executor))
+            .takeUntil((MarshalledEntry<Object, Object> me) -> stopped.get())
```

Comment:
Hrmm tbh I am not exactly sure. I think it has to invoke `doOnNext` before it can call `takeUntil` again. Let me change it and we can find out :)

Reply:
Yeah so it will always only modify on a single thread at a time. I can change it to assert it equals the size exactly then. This should be better, because I could make it so we are doing this across multiple threads with `Flowable.parallel` or `Flowable.flatMap`, but then we are testing rxjava2, which isn't the point of this test.

Although tbh you wouldn't want to do cancellation like this in the first place, you should just invoke `Flowable.take(100)` instead.

The more I thought about it, this method should just be removed 🗡